File: pallycon-drm-sdk/pallycon_drm_sdk_ios/ios/Classes/models/PallyConEventType.swift
//
//  EventType.swift
//  connectivity_plus
//
//  Created by sungju Yun on 2023/01/04.
//

import Foundation

enum PallyConEventType {
    case prepare
    case complete
    case pause
    case remove
    case stop
    case contentDataError
    case drmError
    case licenseServerError
    case downloadError
    case networkConnectedError
    case detectedDeviceTimeModifiedError
    case migrationError
    case unknownError
}

extension PallyConEventType {
    var name: String {
        switch self {
        case .prepare: return "prepare"
        case .complete: return "complete"
        case .pause: return "pause"
        case .remove: return "remove"
        case .stop: return "stop"
        case .contentDataError: return "contentDataError"
        case .drmError: return "drmError"
        case .licenseServerError: return "licenseServerError"
        case .downloadError: return "downloadError"
        case .networkConnectedError: return "networkConnectedError"
        case .detectedDeviceTimeModifiedError: return "detectedDeviceTimeModifiedError"
        case .migrationError: return "migrationError"
        case .unknownError: return "unknownError"
        }
    }
}

File: pallycon-drm-sdk/pallycon_drm_sdk_interface/lib/src/enums/pallycon_event_type.dart
/// Event type of PallyCon SDK Event
enum PallyConEventType {
  /// Download preparation completed
  prepare,

  /// The download completed
  complete,

  /// The download paused
  pause,

  /// The download is removed
  remove,

  /// The download is stop
  stop,

  /// The download is start
  download,

  /// Error when the content information to be downloaded is incorrect
  /// It mainly occurs when the value of the function parameter [PallyConContentConfiguration] is incorrect.
  contentDataError,

  /// License error
  drmError,

  /// Server error when issuing license
  licenseServerError,

  /// Error during download
  downloadError,

  /// Error when there is no network connection
  networkConnectedError,

  /// Error that occurs when the time is forcibly manipulated on an Android device
  detectedDeviceTimeModifiedError,

  /// Error that occurs when migrating from SDK
  migrationError,

  /// Error that occurs when a license cipher from SDK
  licenseCipherError,

  /// Unknown error type
  unknown
}
